libavcodec/vaapi_hevc.c

```diff
@@ -88,19 +88,21 @@ static int find_frame_rps_type(const HEVCContext *h, const HEVCFrame *pic)
             return VA_PICTURE_HEVC_RPS_LT_CURR;
     }
 
+    if (h->ps.pps->pps_curr_pic_ref_enabled_flag)
+        return VA_PICTURE_HEVC_LONG_TERM_REFERENCE;
+
     return 0;
 }
 
 static void fill_vaapi_reference_frames(const HEVCContext *h, VAPictureParameterBufferHEVC *pp)
 {
-    const HEVCFrame *current_picture = h->ref;
     int i, j, rps_type;
 
     for (i = 0, j = 0; i < FF_ARRAY_ELEMS(pp->ReferenceFrames); i++) {
         const HEVCFrame *frame = NULL;
 
         while (!frame && j < FF_ARRAY_ELEMS(h->DPB)) {
-            if (&h->DPB[j] != current_picture && (h->DPB[j].flags & (HEVC_FRAME_FLAG_LONG_REF | HEVC_FRAME_FLAG_SHORT_REF)))
+            if ((h->DPB[j].flags & (HEVC_FRAME_FLAG_LONG_REF | HEVC_FRAME_FLAG_SHORT_REF)))
```

Comment:
&h->DPB[j] != current_picture may use for some special bistream. Remove it may introduce regression. Do we need pps_curr_pic_ref_enabled_flag here?